File: game/Enemy.java
package game;

class Enemy extends Entity {
    public Enemy(String name, int health, int attackPower) {
        super(name, health, attackPower);
    }

    @Override
    public void attack(Entity character) {
        System.out.println(name + " attacks " + character.name + " for " + attackPower + " damage.");
        character.health -= attackPower;
        if (character.health <= 0) {
            System.out.println(character.name + " has been defeated!");
        } else {
            System.out.println(character.name + " has " + character.health + " health remaining.");
        }
    }
}

File: game/Enemy.cs
namespace game
{
    public class Enemy : Entity
    {
        public int ExpValue { get; }
        public Enemy(string name, int health, int attackPower, int expValue)
            : base(name, health, attackPower)
        {
            ExpValue = expValue;
        }

        public void TakeDamage(int damage)
        {
            health -= damage;
            
            if (health < 0)
            {
                health = 0;
            }
            
            Console.WriteLine($"{name} takes {damage} damage.");
            Console.WriteLine($"{name} health: {health}\n");
        }

        public override void Attack(Entity target)
        {
            if (target is Character character)
            {
                Console.WriteLine($"{name} attacks {character.name} for {attackPower} damage.");
                character.TakeDamage(attackPower);
            }
        }
    }
}
